cli/proc_state.rs

```diff
@@ -395,30 +425,12 @@ impl ProcState {
     );
     let maybe_locker = as_maybe_locker(self.lockfile.clone());
     let maybe_imports = self.get_maybe_imports()?;
-    let node_resolver = NodeEsmResolver::new(
-      self.maybe_import_map.clone().map(ImportMapResolver::new),
-    );
-    let maybe_import_map_resolver =
-      self.maybe_import_map.clone().map(ImportMapResolver::new);
-    let maybe_jsx_resolver = self
-      .maybe_config_file
-      .as_ref()
-      .map(|cf| {
-        cf.to_maybe_jsx_import_source_module()
-          .map(|im| JsxResolver::new(im, maybe_import_map_resolver.clone()))
-      })
-      .flatten();
-    let maybe_resolver = if self.flags.compat {
-      Some(node_resolver.as_resolver())
-    } else if maybe_jsx_resolver.is_some() {
-      // the JSX resolver offloads to the import map if present, otherwise uses
-      // the default Deno explicit import resolution.
-      maybe_jsx_resolver.as_ref().map(|jr| jr.as_resolver())
-    } else {
-      maybe_import_map_resolver
-        .as_ref()
-        .map(|im| im.as_resolver())
-    };
+    let maybe_resolver: Option<&dyn deno_graph::source::Resolver> =
+      if let Some(resolver) = &self.maybe_resolver {
+        Some(resolver.as_ref())
+      } else {
+        None
+      };
```

Comment:
Longhand version of saying this?
```suggestion
    let maybe_resolver: Option<&dyn deno_graph::source::Resolver> =
      self.maybe_resolver.as_ref();
```
(Or maybe `.as_deref()`. Likewise around line 661.)

Reply:
I tried that before but it doesn't work:
```
error[E0308]: mismatched types
   --> cli/proc_state.rs:429:7
    |
428 |     let maybe_resolver: Option<&dyn deno_graph::source::Resolver> =
    |                         ----------------------------------------- expected due to this
429 |       self.maybe_resolver.as_ref();
    |       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected trait object `dyn Resolver`, found struct `Arc`
    |
    = note: expected enum `std::option::Option<&dyn Resolver>`
               found enum `std::option::Option<&Arc<dyn Resolver + std::marker::Send + std::marker::Sync>>`

```

I had to add additional trait boundaries (`Send + Sync`) to the `dyn Resolver` and found that it's the only method to get it past type checking.